### PR TITLE
Add more SetParam tests for Configuration

### PR DESCRIPTION
**Pull request by ami-GS**
## Description

SetParam Tests for Configuration #1215

## Testing

This is test

## Documentation

N/A

### CODE REVIEW

File: src/core/quicdef.h [hunk 1]
```diff
@@ -490,6 +490,10 @@ CXPLAT_STATIC_ASSERT(
 //
 #define QUIC_CONGESTION_CONTROL_ALGORITHM_DEFAULT   QUIC_CONGESTION_CONTROL_ALGORITHM_CUBIC
 
+#define QUIC_TP_MAX_ACK_DELAY_DEFAULT                       25 // ms
+#define QUIC_TP_MAX_ACK_DELAY_MAX                           ((1 << 14) - 1)
+#define QUIC_TP_MIN_ACK_DELAY_MAX                           ((1 << 24) - 1)
```
Comment (nibanks) [inline src/core/quicdef.h:495]: Let's go ahead and move all the defines from transport_params.h to here.

File: src/test/lib/CMakeLists.txt [hunk 2]
```diff
@@ -21,7 +21,9 @@ set(SOURCES
 
 add_library(testlib STATIC ${SOURCES})
 
-target_include_directories(testlib PRIVATE ${PROJECT_SOURCE_DIR}/src/test)
+target_include_directories(testlib
+    PRIVATE ${PROJECT_SOURCE_DIR}/src/test
+    PRIVATE ${PROJECT_SOURCE_DIR}/src/core)
```
Comment (nibanks) [inline src/test/lib/CMakeLists.txt:26]: I wonder if it's worth moving `quicdef.h` to `.\inc\quic_defines.h`.

Comment (ami-GS) [reply src/test/lib/CMakeLists.txt:26]: hmm, it is difficult for me to decide. it should be different PR if needed

File: src/core/quicdef.h (hunk 1 above)
Comment (ami-GS) [reply src/core/quicdef.h:495]: will do for base branch (_global). then rebase this branch on it.

**Review by nibanks — DISMISSED** [review]

**Comment by nibanks** [comment]
Looks like all the tests are failing.

**Comment by ami-GS** [comment]
Can we ignore?

**Comment by nibanks** [comment]
> Can we ignore?
> 

We need to figure out what/why first.

File: src/core/quicdef.h [hunk 3]
```diff
@@ -538,6 +538,54 @@ CXPLAT_STATIC_ASSERT(
 //
 #define QUIC_TP_MAX_STREAMS_MAX                             ((1ULL << 60) - 1)
 
+/*************************************************************
```
Comment (nibanks) [inline src/core/quicdef.h:541]: Please remove this duplicate/unnecessary change.

File: src/test/lib/ApiTest.cpp [hunk 4]
```diff
@@ -2074,7 +2067,7 @@ void QuicTestGlobalSetParam()
             QUIC_STATUS_INVALID_PARAMETER,
             MsQuic->SetParam(
                 nullptr,
-                QUIC_PARAM_GLOBAL_LIBRARY_VERSION,
+                QUIC_PARAM_GLOBAL_LIBRARY_GIT_HASH,
```
Comment (nibanks) [inline src/test/lib/ApiTest.cpp:2070]: Accidental change?

File: src/test/lib/precomp.h [hunk 5]
```diff
@@ -11,6 +11,8 @@
 #define QUIC_DEFAULT_CLIENT_CRED_FLAGS \
     (QUIC_CREDENTIAL_FLAG_CLIENT | QUIC_CREDENTIAL_FLAG_NO_CERTIFICATE_VALIDATION)
 
+#define QUIC_TEST_APIS 1
+
```
Comment (nibanks) [inline src/test/lib/precomp.h:15]: Is this really necessary?

**Review by nibanks — CHANGES_REQUESTED** [review]

File: src/core/quicdef.h (hunk 3 above)
Comment (ami-GS) [reply src/core/quicdef.h]: PR depends on _global branch made duplicate change by accident...

File: src/test/lib/ApiTest.cpp (hunk 4 above)
Comment (ami-GS) [reply src/test/lib/ApiTest.cpp]: Accidental🤦‍♂️

File: src/test/lib/precomp.h (hunk 5 above)
Comment (ami-GS) [reply src/test/lib/precomp.h]: forget to remove

**Review by nibanks — APPROVED** [review]
